File: 1001-1100/1013-Partition Array Into Three Parts With Equal Sum/1013-Partition Array Into Three Parts With Equal Sum.py
class Solution:
    def canThreePartsEqualSum(self, A: List[int]) -> bool:
        total = sum(A)
        if total % 3:
            return False
        target = total // 3
        curr = count = 0
        tc = 2 if target else 3
        for a in A:
            curr += a
            if curr == target:                
                count += 1
                if count == tc:
                    return True
                curr = 0
        return False

File: 1001-1100/1013-Partition Array Into Three Parts With Equal Sum/1013-Partition Array Into Three Parts With Equal Sum.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& A) {
        int total = accumulate(A.begin(), A.end(), 0);
        if (total % 3) {
            return false;
        }
        int target = total / 3;
        int curr = 0, count = 0, tc = target ? 3 : 2;
        for (int a : A) {
            curr += a;
            if (curr == target) {
                if (++count == tc) {
                    return true;
                }
                curr = 0;
            }
        }
        
        return false;
    }
};
